src/manage/approveAccessRequest.ts

```diff
@@ -161,7 +141,92 @@ export async function approveAccessRequest(
     options
   );
 
-  return issueAccessVc(resourceOwner, grantBody, options);
+  return issueAccessVc(grantBody, options);
+}
+
+/**
+ * Approve an access request. The content of the approved access request is provided
+ * as a Verifiable Credential which properties may be overriden if necessary.
+ *
+ * @param requestVc The Verifiable Credential representing the Access Request. If
+ * not conform to an Access Request, the function will throw.
+ * @param requestOverride Elements overriding information from the provided Verifiable Credential.
+ * @param options Optional properties to customise the access grant behaviour.
+ * @returns A Verifiable Credential representing the granted access.
+ * @since 0.0.1.
+ */
+export async function approveAccessRequest(
+  // If the VC is specified, all the overrides become optional
+  requestVc: VerifiableCredential | URL | UrlString,
+  requestOverride?: Partial<ApproveAccessRequestOverrides>,
+  options?: AccessBaseOptions
+): Promise<VerifiableCredential>;
+/**
+ * @deprecated Please remove the `resourceOwner` parameter.
+ */
+export async function approveAccessRequest(
+  resourceOwner: WebId,
+  // If the VC is specified, all the overrides become optional
+  requestVc: VerifiableCredential | URL | UrlString,
+  requestOverride?: Partial<ApproveAccessRequestOverrides>,
+  options?: AccessBaseOptions
+): Promise<VerifiableCredential>;
+/**
```

Comment:
Could we consider removing deprecated functions and releasing a minor?

We're still at 0.x and I don't think it makes sense to try and be forward compatible here. It polutes the code (and only pod browser is really consuming this)... Even not considering the lack of consumers, I think a 0.x library should generally really remain clean and remove obsolete function signatures as soon as they become such.

Is there even a live VC issuer that would accept the result of such function calls?

Reply:
I think I'd be inclined to agree that we can just deprecate and bump minor: https://jubianchi.github.io/semver-check/#/^0.4.1/0.5.0 (though the text there on the constraint is confusing) — pod browser depends on `^0.4.1` which won't accidentally upgrade to `0.5.0` unlike say `^1.2.0` matching up to, but not including, `2.0.0-0`

Reply:
(I was pretty confused reading this code earlier and wasn't sure how the docs would end up with this change)

Reply:
Initially, the VC subject was included in the request but ignored by the issuer. Now, the deprecated signature maps to the correct function call ignoring the deprecated parameter, so the VC issuer wouldn't see any difference.

Reply:
I'll double check how the docs look indeed, I was wondering about that too.